api/src/paths/project/{projectId}/publish.ts

```diff
@@ -0,0 +1,130 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { SYSTEM_ROLE } from '../../../constants/roles';
+import { getDBConnection } from '../../../database/db';
+import { HTTP400, HTTP500 } from '../../../errors/CustomError';
+import { projectIdResponseObject } from '../../../openapi/schemas/project';
+import { getLogger } from '../../../utils/logger';
+import { logRequest } from '../../../utils/path-utils';
+import { updateProjectPublishStatusSQL } from '../../../queries/project/project-update-queries';
+
+const defaultLog = getLogger('paths/project/{projectId}/publish');
+
+export const PUT: Operation = [logRequest('paths/project/{projectId}/publish', 'PUT'), publishProject()];
+
+PUT.apiDoc = {
+  description: 'Publish a project.',
+  tags: ['project'],
+  security: [
+    {
+      Bearer: [SYSTEM_ROLE.SYSTEM_ADMIN, SYSTEM_ROLE.PROJECT_ADMIN]
+    }
+  ],
+  parameters: [
+    {
+      in: 'path',
+      name: 'projectId',
+      schema: {
+        type: 'number'
+      },
+      required: true
+    }
+  ],
+  requestBody: {
+    description: 'Publish put request object.',
+    content: {
+      'application/json': {
+        schema: {
+          title: 'Publish request object',
+          type: 'object',
+          required: ['publish'],
+          properties: {
+            publish: {
+              title: 'publish?',
+              type: 'boolean'
```

Comment:
is this accurate? I thought you are passing the `publish_timestamp`

Reply:
Yeah I changed it to boolean .. because when we unpublish, we need to send either an empty string or null.  It's just as easy for the DB to create the timestamp ... so I went with that.

Reply:
We definitely want the timestamp to be created by the server.  So the endpoint can either take a boolean (and support both publish/unpublish), or we can have to separate endpoints (one for publish, one for unpublish), and the user doesnt have to provide any data (other than the project ID, but that is already part of the route).